### PR TITLE
Fix probes

### LINKED ISSUE #732: Add liveness probes to deployments
Currently, there are no liveness and readiness probes in SC4SNMP Kubernetes infrastructure.
Would be beneficial to add it.

### PR DESCRIPTION
**Pull request by pgrenda-splunk**
# Description
Add kubernetes liveness and readiness probes to project configuration.
Fixes #732

## Type of change

Please delete options that are not relevant.

- [ X ] improvement

## How Has This Been Tested?

Celery Worker Tests:

Readiness Probe Test:

Verified creation of /tmp/worker_ready.
Removed /tmp/worker_ready.
Checked microk8s kubectl describe pod <name> -n sc4snmp for readiness probe trigger information.
Liveness Probe Test:

Since the liveness probe is created with every heartbeat, it can be removed quickly with the following: 
```
while true; do rm /tmp/worker_heartbeat; sleep 0.1; done
```
Checked microk8s kubectl describe pod <name> -n sc4snmp for liveness probe trigger information.
Verified if pod status changed to "completed" and restart counter increased by 1.

## Checklist

- [ X ] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ X ] I have run pre-commit on all files before creating the PR
- [ X ] I have commented my code, particularly in hard-to-understand areas
- [ X ] I have made corresponding changes to the documentation
- [ X ] I have added tests that prove my fix is effective or that my feature works
- [ X ] New and existing unit tests pass locally with my changes
- [ X ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: charts/splunk-connect-for-snmp/Chart.lock [hunk 1]
```diff
@@ -7,6 +7,6 @@ dependencies:
   version: 17.3.18
 - name: mibserver
   repository: https://pysnmp.github.io/mibs/charts/
-  version: 1.15.1
-digest: sha256:9e01a49851f1ea1ba6b1c75785ac2b9a03ab5142588ef8a071f694d904c0c1f1
-generated: "2023-02-09T15:52:42.216766+01:00"
+  version: 1.15.2
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/Chart.lock:10]: maybe let's not override the helm chart version here, would be good to open another PR specifically for this purpose

File: charts/splunk-connect-for-snmp/values.yaml [hunk 2]
```diff
@@ -455,8 +537,64 @@ mongodb:
...
+    exec:
+      # The exec command for the liveness probe to run in the container.
+      command:
+        - mongo
+        - --eval
+        - 'db.adminCommand("ping")'
+    # Number of seconds after the container has started before liveness probes are initiated.
+    initialDelaySeconds: 30
+    # How often (in seconds) to perform the probe.
+    periodSeconds: 10
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/values.yaml:551]: Have you checked/read how it affects the performance of MongoDB?

File: splunk_connect_for_snmp/poller.py [hunk 3]
```diff
@@ -55,25 +50,6 @@
 INVENTORY_PATH = os.getenv("INVENTORY_PATH", "/app/inventory/inventory.csv")
 INVENTORY_REFRESH_RATE = int(os.getenv("INVENTORY_REFRESH_RATE", "600"))
 
-
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp/poller.py:58]: can you rebase to develop? it sees the code that doesn't exists as a HEAD

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]
Can you also create new case in rendered manifests and run `make render`?

File: charts/splunk-connect-for-snmp/values.yaml [hunk 4]
```diff
@@ -455,8 +537,64 @@ mongodb:
...
+      - 'db.adminCommand("ping")'
+    # Number of seconds after the container has started before readiness probes are initiated.
+    initialDelaySeconds: 10
+    # How often (in seconds) to perform the probe.
+    periodSeconds: 5
 redis:
   architecture: standalone
   auth:
     enabled: false
+  livenessProbe:
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/values.yaml:572]: same with redis: https://github.com/bitnami/charts/tree/main/bitnami/redis

File: charts/splunk-connect-for-snmp/values.yaml (hunk 2 above)
Comment (pgrenda-splunk) [reply charts/splunk-connect-for-snmp/values.yaml]: Yes you are complete right. I just placed livenessProbe: enabled flag there to decapsulated it from underlaying layers :)

File: charts/splunk-connect-for-snmp/values.yaml (hunk 4 above)
Comment (pgrenda-splunk) [reply charts/splunk-connect-for-snmp/values.yaml:553]: same changes as in mongo case

File: charts/splunk-connect-for-snmp/templates/worker/sender/deployment.yaml [hunk 5]
```diff
@@ -46,6 +46,22 @@ spec:
           env:
             {{- include "environmental-variables" . | nindent 12 }}
             {{- include "environmental-variables-sender" . | nindent 12 }}
+          {{- if .Values.worker.poller.livenessProbe }}
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/templates/worker/sender/deployment.yaml:49]: here it should be worker.sender

File: charts/splunk-connect-for-snmp/templates/worker/trap/deployment.yaml [hunk 6]
```diff
@@ -59,6 +59,23 @@ spec:
               readOnly: false
           resources:
             {{- toYaml .Values.worker.trap.resources | nindent 12 }}
+      {{- if .Values.worker.trap.livenessProbe }}
+      livenessProbe:
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/templates/worker/trap/deployment.yaml:63]: you don't need to use toYaml everywhere, just cast .Values.worker.trap.livenessProbe toYaml

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]

**Review by omrozowicz-splunk — COMMENTED** [review]
I'd add one small thing - enabled parameter in main values.yaml to let user know how to enable and disable it

File: charts/splunk-connect-for-snmp/values.yaml [hunk 7]
```diff
@@ -273,6 +272,40 @@ worker:
         # the resources requests for poller worker container
       requests:
         cpu: 250m
+      # Liveness probes are used in Kubernetes to know when a pod is alive or dead.
```
Comment (omrozowicz-splunk) [inline charts/splunk-connect-for-snmp/values.yaml:275]: too many spaces here

**Review by omrozowicz-splunk — CHANGES_REQUESTED** [review]

**Review by omrozowicz-splunk — APPROVED** [review]
